src/meta/proto-conv/tests/it/proto_conv.rs

```diff
@@ -412,57 +413,118 @@ fn test_load_old() -> anyhow::Result<()> {
+                1, 10, 35, 10, 9, 116, 105, 109, 101, 115, 116, 97, 109, 112, 26, 16, 114, 8, 8, 5,
+                160, 6, 10, 168, 6, 1, 160, 6, 10, 168, 6, 1, 160, 6, 10, 168, 6, 1, 10, 24, 10, 6,
+                115, 116, 114, 105, 110, 103, 26, 8, 122, 0, 160, 6, 10, 168, 6, 1, 160, 6, 10,
+                168, 6, 1, 10, 61, 10, 6, 115, 116, 114, 117, 99, 116, 26, 45, 130, 1, 36, 10, 3,
+                102, 111, 111, 10, 3, 98, 97, 114, 18, 8, 18, 0, 160, 6, 10, 168, 6, 1, 18, 8, 122,
+                0, 160, 6, 10, 168, 6, 1, 160, 6, 10, 168, 6, 1, 160, 6, 10, 168, 6, 1, 160, 6, 10,
+                168, 6, 1, 10, 40, 10, 5, 97, 114, 114, 97, 121, 26, 25, 138, 1, 16, 10, 8, 18, 0,
+                160, 6, 10, 168, 6, 1, 160, 6, 10, 168, 6, 1, 160, 6, 10, 168, 6, 1, 160, 6, 10,
+                168, 6, 1, 10, 32, 10, 7, 118, 97, 114, 105, 97, 110, 116, 26, 15, 146, 1, 6, 160,
+                6, 10, 168, 6, 1, 160, 6, 10, 168, 6, 1, 160, 6, 10, 168, 6, 1, 10, 38, 10, 13,
+                118, 97, 114, 105, 97, 110, 116, 95, 97, 114, 114, 97, 121, 26, 15, 154, 1, 6, 160,
+                6, 10, 168, 6, 1, 160, 6, 10, 168, 6, 1, 160, 6, 10, 168, 6, 1, 10, 39, 10, 14,
+                118, 97, 114, 105, 97, 110, 116, 95, 111, 98, 106, 101, 99, 116, 26, 15, 162, 1, 6,
+                160, 6, 10, 168, 6, 1, 160, 6, 10, 168, 6, 1, 160, 6, 10, 168, 6, 1, 10, 35, 10, 8,
+                105, 110, 116, 101, 114, 118, 97, 108, 26, 17, 170, 1, 8, 8, 2, 160, 6, 10, 168, 6,
+                1, 160, 6, 10, 168, 6, 1, 160, 6, 10, 168, 6, 1, 18, 6, 10, 1, 97, 18, 1, 98, 160,
+                6, 10, 168, 6, 1, 34, 10, 40, 97, 32, 43, 32, 50, 44, 32, 98, 41, 42, 10, 10, 3,
+                120, 121, 122, 18, 3, 102, 111, 111, 50, 2, 52, 52, 58, 10, 10, 3, 97, 98, 99, 18,
+                3, 100, 101, 102, 64, 0, 74, 10, 40, 97, 32, 43, 32, 50, 44, 32, 98, 41, 82, 7,
+                100, 101, 102, 97, 117, 108, 116, 162, 1, 23, 50, 48, 49, 52, 45, 49, 49, 45, 50,
+                56, 32, 49, 50, 58, 48, 48, 58, 48, 57, 32, 85, 84, 67, 170, 1, 23, 50, 48, 49, 52,
+                45, 49, 49, 45, 50, 57, 32, 49, 50, 58, 48, 48, 58, 49, 48, 32, 85, 84, 67, 178, 1,
+                13, 116, 97, 98, 108, 101, 95, 99, 111, 109, 109, 101, 110, 116, 186, 1, 6, 160, 6,
+                10, 168, 6, 1, 202, 1, 1, 99, 202, 1, 1, 99, 202, 1, 1, 99, 202, 1, 1, 99, 202, 1,
+                1, 99, 202, 1, 1, 99, 202, 1, 1, 99, 202, 1, 1, 99, 202, 1, 1, 99, 202, 1, 1, 99,
+                202, 1, 1, 99, 202, 1, 1, 99, 202, 1, 1, 99, 202, 1, 1, 99, 202, 1, 1, 99, 202, 1,
+                1, 99, 202, 1, 1, 99, 202, 1, 1, 99, 202, 1, 1, 99, 202, 1, 1, 99, 202, 1, 1, 99,
+                160, 6, 10, 168, 6, 1,
+            ];
+            let p: pb::TableMeta = common_protos::prost::Message::decode(tbl_meta_v10.as_slice())
+                .map_err(print_err)?;
+
+            let got = mt::TableMeta::from_pb(p).map_err(print_err)?;
+
+            let want = new_table_meta();
```

Comment:
There should be a dedicated instance, but not the latest version of table-meta,  for testing this `v10` table-meta.

And using some value other than the default value `"default"` for the compatibility test would be better, to assert that the field `catalog` is loaded from the message but not just filled with a default value.

Reply:
Addressed in https://github.com/datafuselabs/databend/pull/7835/commits/c5c12c35e2f47af361a2f6a618b68728fb2ea4b1